core/src/main/java/apoc/trigger/Trigger.java

```diff
@@ -50,28 +51,64 @@ public TriggerInfo( String name, String query, Map<String,Object> selector, Map<
 
     @Context public GraphDatabaseService db;
 
-    static TriggerUtils triggerUtils = new TriggerUtils();
+    @Context public TriggerHandler triggerHandler;
+    
+    @Context public Log log;
+
+    private void preprocessDeprecatedProcedures() {
+        final String msgDeprecation =
+                "use the `apoc.trigger.install`, `apoc.trigger.drop`, `apoc.trigger.dropAll`, `apoc.trigger.stop`, and `apoc.trigger.start` procedures \n" + 
+                "instead of, respectively, `apoc.trigger.add`, `apoc.trigger.remove`, `apoc.trigger.removeAll`, `apoc.trigger.pause`, and `apoc.trigger.resume`.";
+                
+        log.warn("Please note that the current procedure is deprecated, \n" + msgDeprecation);
+        
+        if (!Util.isWriteableInstance(db, GraphDatabaseSettings.SYSTEM_DATABASE_NAME)) {
+            throw new RuntimeException("It's not possible to write into a cluster member with a non-LEADER system database.\n" +
+                    "Either the procedure using the bolt against a core protocol with LEADER system database, \n" +
```

Comment:
The wording of this message needs to be changed

`Either use the procedure executing it with bolt against a LEADER system database`

Reply:
Are we sure the procedure is not redirected by Fabric to a user LEADER even if we ask for system LEADER? I think we should recommend using the other procedures directly, not using these ones at all in this case.

Reply:
Changed it